tensorflow_lite_support/cc/task/processor/image_preprocessor.cc

```diff
@@ -68,6 +68,14 @@ absl::Status ImagePreprocessor::Init(
         "ImagePreprocessor only supports RGB color space for now.");
   }
 
+  // Determine if the input shape is resizable.
+  const TfLiteIntArray* dims_signature = Tensor()->dims_signature;
+
+  if (dims_signature != nullptr) {
+    // Only the HxW dimensions support mutability.
+    is_height_mutable_ = dims_signature->data[1] == -1;
+    is_width_mutable_ = dims_signature->data[2] == -1;
+  }
```

Comment:
@lu-wang-g  I was thinking, we can allow batch, depth to be mutable. If, however, the values of batch and depth are invalid (not 1, 3 respectively) it will be caught in **BuildInputSpecs()** method. What's your thoughts?

Reply:
But BuildInputImageTensorSpecs() is only invoked in Init(). It won't catch it If batch and depth changed in ImagePreprocessor::Preprocess(), right?

Reply:
I have a question, why was this not checked before? I mean the user can do the following

```c++
// Create engine and preprocessor.
auto engine_ = absl::make_unique<TfLiteEngine>();
engine_->BuildandInit(); // Builds + init.
auto preprocessor = ImageProcessor::Create(..);

// Re-size tensor
engine_->ResizeInputTensor(1, 420, 380, 5); // Batch x height x width x channel. Notice batch and channel are invalid

// Undefined behaviour??
engine_->Preprocess(*buffer);
```

I checked `Preprocess` it claims the following:

```
    // Preprocess input image to fit model requirements.
    // For now RGB is the only color space supported, which is ensured by
    // `InitInternal`.
```

but we know that this will fail if we change after Init() right? Just as you mentioned.

Reply:
So, only `ResizeInputTensor`, but `ResizeInputTensorStrict` would be exposed via `engine_`, correct?

Reply:
Also the comment "For now RGB is only color space support, which is ensured by 'InitInternal'" is redundant.

Reply:
`ResizeInputTensorStrict` will not be exposed through `engine_wrapper_` either. But processors will have access to engine, you can still do  `ResizeInputTensorStrict` within processors.